.github/workflows/ci.yml

```diff
@@ -315,6 +315,38 @@ jobs:
           path: htmlcov
           include-hidden-files: true
 
+  devcontainer:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Free Disk Space on GitHub Actions runner
+        # Runners come with many preinstalled tools that we don't need for our build.
+        # This action removes them to free up disk space for the following heavy devcontainer build.
+        uses: jlumbroso/free-disk-space@v1.3.1
+        with:
+          tool-cache: false
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: true
+          swap-storage: true
+
+      - uses: actions/checkout@v4
+
+      - name: Build and run dev container task
+        uses: devcontainers/ci@v0.3
+        with:
+          runCmd: |
+            bash -c '
+              uv --version
+              uvx --version
+              pre-commit --version
+              make lint
+              make typecheck
+              make test
+              echo "Dev container built and run successfully"
```

Comment:
As it stands, this CI step will always succeed, even if any of the steps above failed.

For example, on this PR the output currently shows this error, even though the step shows as green:

```
Coverage failure: total of 99.99 is less than fail-under=100.00
```

- Can we please fix this job to fail when any of the steps fail?
- Can we get coverage to 100%? (or mark the uncovered lines as `pragma: lax no cover` if we can't)

Reply:
The CI job can now fail, see `set -o errexit`. Coverage is now 100%.